src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -27,11 +31,30 @@ class AuthenticationFilter : ContainerRequestFilter {
         verify(jwtCookie, requestContext)
     }
 
-    fun verify(cookie: String?, requestContext: ContainerRequestContext) =
+
+    fun verify(cookie: String?, requestContext: ContainerRequestContext) {
+        if (cookie == null) {
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).build())
+            return
+        }
         try {
-            Jwts.parserBuilder().setSigningKey(SECRET_KEY).build().parseClaimsJws(cookie)
+            val claims: Claims = Jwts.parserBuilder()
+                .setSigningKey(SECRET_KEY)
+                .build()
+                .parseClaimsJws(cookie)
+                .body
+
+            val employeeData = EmployeeData(
+                id = (claims["id"] as Number).toLong(),
+                role = Role.fromDatabaseValue(claims["role"] as String),
+                companyId = (claims["companyId"] as Number).toLong(),
+                department = Department.fromDatabaseValue(claims["department"] as String)
+            )
+            requestContext.setProperty(EMPLOYEE, employeeData)
         } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
instead of printing like this, please create a logger as a member of the class:
```
private val logger = org.slf4j.LoggerFactory.getLogger(AuthenticationFilter::class.java)
```

and use it to log the exception:
```
logger.error(e.getMessage, e)
```